siphon/catalog.py

```diff
@@ -410,15 +410,18 @@ def make_access_urls(self, catalog_url, all_services, metadata=None):
             The top level server url
         all_services : List[SimpleService]
             list of :class:`SimpleService` objects associated with the dataset
-        metadata : TDSCatalogMetadata
+        metadata : dict
             Metadata from the :class:`TDSCatalog`
 
         """
-        all_service_dict = {service.name: service for service in all_services}
-        service_name = None
-        if metadata:
-            if 'serviceName' in metadata:
-                service_name = metadata['serviceName']
+        all_service_dict = {}
+        for service in all_services:
+            all_service_dict[service.name] = service
```

Comment:
Should this be in an else? (i.e. should the compound service be in the dictionary or just it's sub services?)

Reply:
You want both. I initially had it as an `else`, and it (rightly) failed tests.

Reply:
👍